hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/lifecycle/OMLifecycleConfigurationCreateRequest.java

```diff
@@ -0,0 +1,169 @@
+        .setCreateLifecycleConfigurationRequest(newCreateRequest.build())
+        .build();
+  }
+
+  @Override
+  public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, ExecutionContext context) {
+    final long transactionLogIndex = context.getIndex();
+    OMMetadataManager metadataManager = ozoneManager.getMetadataManager();
+
+    CreateLifecycleConfigurationRequest createLifecycleConfigurationRequest =
+        getOmRequest().getCreateLifecycleConfigurationRequest();
+
+    LifecycleConfiguration lifecycleConfiguration =
+        createLifecycleConfigurationRequest.getLifecycleConfiguration();
+
+    String volumeName = lifecycleConfiguration.getVolume();
+    String bucketName = lifecycleConfiguration.getBucket();
+
+    OMResponse.Builder omResponse = OmResponseUtil.getOMResponseBuilder(
+        getOmRequest());
+
+    AuditLogger auditLogger = ozoneManager.getAuditLogger();
+    UserInfo userInfo = getOmRequest().getUserInfo();
+
+    String bucketKey = metadataManager.getBucketKey(volumeName, bucketName);
+    IOException exception = null;
+    boolean acquiredBucketLock = false;
+    OMClientResponse omClientResponse = null;
+    Map<String, String> auditMap = new HashMap<>();
+
+    try {
+      if (ozoneManager.getAclsEnabled()) {
+        checkAcls(ozoneManager, OzoneObj.ResourceType.BUCKET, OzoneObj.StoreType.OZONE,
+            IAccessAuthorizer.ACLType.ALL, volumeName, bucketName, null);
+      }
+
+      OmLifecycleConfiguration omLifecycleConfiguration =
+          OmLifecycleConfiguration.getFromProtobuf(lifecycleConfiguration);
+      auditMap = omLifecycleConfiguration.toAuditMap();
+
+      mergeOmLockDetails(metadataManager.getLock()
+          .acquireWriteLock(BUCKET_LOCK, volumeName, bucketName));
+      acquiredBucketLock = getOmLockDetails().isLockAcquired();
+      omLifecycleConfiguration.valid();
+
+      if (!metadataManager.getBucketTable().isExist(bucketKey)) {
+        LOG.debug("bucket: {} in volume: {} doesn't exist", bucketName,
+            volumeName);
+        throw new OMException("Bucket doesn't exist", BUCKET_NOT_FOUND);
+      }
+      omLifecycleConfiguration.setUpdateID(transactionLogIndex);
+
+      metadataManager.getLifecycleConfigurationTable().addCacheEntry(
+          new CacheKey<>(bucketKey),
+          CacheValue.get(transactionLogIndex, omLifecycleConfiguration));
+
+      omResponse.setCreateLifecycleConfigurationResponse(
+          CreateLifecycleConfigurationResponse.newBuilder().build());
+
+      omClientResponse = new OMLifecycleConfigurationCreateResponse(
+          omResponse.build(), omLifecycleConfiguration);
+    } catch (IOException ex) {
+      exception = ex;
+      omClientResponse = new OMLifecycleConfigurationCreateResponse(
+          createErrorOMResponse(omResponse, exception));
+    } finally {
+      if (acquiredBucketLock) {
+        metadataManager.getLock().releaseWriteLock(BUCKET_LOCK, volumeName,
```

Comment:
mergeOmLockDetails() call is missing here.

Reply:
Added.

Reply:
This is missed in set/delete request. 
```
if (omClientResponse != null) {
        omClientResponse.setOmLockDetails(getOmLockDetails());
      }
```